### PR TITLE
Add studio logs settings sidebar

### PR DESCRIPTION
**Pull request by defunctzombie**
**User-Facing Changes**
None.

There is a feature flag to enable this sidebar during studio development.

**Description**
The studio logs settings sidebar is a studio developer feature. It provides an interface for enabling/disabling specific logging components and groups of components.

To use the logs control sidebar you must enable "Studio Debug Panels" in the app settings. This feature is enabled by default in development.

Fixes: #4384

---
Screen Shot 2022-09-12 at 10 33 59 AM

### CODE REVIEW

**Comment by defunctzombie** [comment]
@esthersweon We can delete the docs for https://foxglove.dev/docs/studio/panels/studio-logs. Even without this PR I think we should delete those docs. They are not helping any user.

**Review by foxymiles — APPROVED** [review]

**Review by jtbandes — APPROVED** [review]

**Review by snosenzo — APPROVED** [review]
not a big fan of going from checkbox to `eye`cons, but otherwise looks good
I think first impression of the panel isn't clear that you can toggle things on/off without hovering over them

**Comment by snosenzo** [comment]
Also, should there be some instructional text at the top about where to find these logs?

**Comment by defunctzombie** [comment]
> not a big fan of going from checkbox to `eye`cons, but otherwise looks good
> I think first impression of the panel isn't clear that you can toggle things on/off without hovering over them

That's a property of our settings ui. I opted to stay consistent with that api so we use what our users are experiencing. Maybe this makes us think about this more :D

File: packages/studio-base/src/Workspace.tsx [hunk 1]
```diff
@@ -105,7 +106,8 @@ type SidebarItemKey =
   | "account"
   | "layouts"
   | "preferences"
-  | "help";
+  | "help"
+  | "studio-logs-control";
```
[the remark this answers is not in the document]
Comment (defunctzombie) [reply packages/studio-base/src/Workspace.tsx:110]: `studio-logs` seemed too generic to me. This is specifically about controlling what is on/off. Is there a better word for it?

[the remark this answers is not in the document]
Comment (jhurliman) [reply packages/studio-base/src/Workspace.tsx:110]: I would either change `preferences` to `preferences-control` or `studio-logs-control` to `studio-logs` for consistency between these two sidebars that toggle a series of things on or off.

**Comment by amacneil** [comment]
> > not a big fan of going from checkbox to eyecons, but otherwise looks good
I think first impression of the panel isn't clear that you can toggle things on/off without hovering over them
> 
> That's a property of our settings ui. I opted to stay consistent with that api so we use what our users are experiencing. Maybe this makes us think about this more :D

Yeah, I have the same complaint for the 3D panel, so let's keep things consistent and we can discuss changing the icons separately.

File: packages/studio-base/src/theme/icons.tsx [hunk 2]
```diff
@@ -41,6 +41,7 @@ import {
   OpenFileIcon,
   RectangularClippingIcon,
   Variable2Icon,
+  BacklogListIcon,
```
Comment (2metres) [inline packages/studio-base/src/theme/icons.tsx:44]: 🍹 Alphabetize pls

Comment (defunctzombie) [reply packages/studio-base/src/theme/icons.tsx]: done

**Pull request merged**